### PR TITLE
feat(code-snippet): sync with @carbon/react v11

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#10119 
### Description

Sync code-snippet with @carbon/react

### Changelog

**New**

- refactored code-snippet
- added new buttonClasses to `cds-button`

**Changed**

- updated stories to align

### CODE REVIEW

**Review by RichKummer — APPROVED** [review]
LGTM!

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.scss [hunk 1]
```diff
@@ -21,140 +21,121 @@ $css--plex: true !default;
...
-// TODO: deprecate light color scheme
-// :host(#{$prefix}-code-snippet[color-scheme='light']) {
-//   background-color: $layer-02;
-
-//   button.#{$prefix}--btn.#{$prefix}--snippet-btn--expand {
-//     background-color: $layer-02;
-//   }
+#{$prefix}-copy {
+  code {
+    padding: 0 0.5rem;
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/code-snippet/code-snippet.scss:67]:
```suggestion
    padding: 0 $spacing-03;
```

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.scss [hunk 2]
```diff
@@ -21,140 +21,121 @@ $css--plex: true !default;
...
-      white-space: pre;
-      padding-right: 0;
-
-      &::after {
-        // Use top-level pseudo element instead because `<pre>` here is inside a scrolling container
-        content: none;
-      }
+  #{$prefix}-button {
+    .#{$prefix}--snippet__icon {
+      margin-left: 0.5rem;
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/code-snippet/code-snippet.scss:102]:
```suggestion
      margin-left: $spacing-03;
```

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.ts [hunk 3]
```diff
@@ -79,28 +47,51 @@ const renderCode = ({
...
+   * `true` if show more or show less btn is visible
    */
-  private _showExpando = false;
+  private _shouldShowMoreLessBtn = false;
 
   /**
    * Handles `click` event on the copy button.
    */
-  private _handleClickCopyButton() {
+  private _handlCopyClick() {
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/code-snippet/code-snippet.ts:84]:
```suggestion
  private _handleCopyClick() {
```

minor typo

**Review by IgnacioBecerra — COMMENTED** [review]
Everything looks good! Just a couple comments:

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.ts (hunk 3 above)
Comment (ariellalgilmore) [reply packages/carbon-web-components/src/components/code-snippet/code-snippet.ts]: fixed!

File: packages/carbon-web-components/src/components/code-snippet/code-snippet.scss [hunk 4]
```diff
@@ -21,140 +21,121 @@ $css--plex: true !default;
...
-:host(#{$prefix}-code-snippet[type='multi']),
-:host(#{$prefix}-code-snippet-skeleton[type='multi']) {
-  @extend .#{$prefix}--snippet--multi;
+    @extend .#{$prefix}--snippet--multi, .#{$prefix}--snippet-container;
+  }
 
-  height: auto;
+  pre {
+    margin: 0;
+    padding: 0 $spacing-08 rem(24px) 0;
```
Comment (IgnacioBecerra) [inline packages/carbon-web-components/src/components/code-snippet/code-snippet.scss:86]:
```suggestion
    padding: 0 $spacing-08 $spacing-06 0;
```

**Review by IgnacioBecerra — APPROVED** [review]
LGTM! I think I missed this one padding value in my last review, but everything else is good!

**Pull request merged**